### DISCUSSION TITLE
Does password hashing have to be run on the same server that code-server is running on

### DISCUSSION THREAD

**Discussion by simonjcarr** [coder/code-server#4684] [Q&A]
I am trying to run code-server in docker. I want to hash my password using the Python argon2 module and use the generated password for vm's running on Linux.

When I fire up the containers, I am told that a password has been supplied using $HASSHED_PASSWORD, but I always get the error incorrect password.

This is how I am creating the password

```
➜  ~ echo -n "password" | npx argon2-cli -e
$argon2i$v=19$m=4096,t=3,p=1$pYZYdaYLyi6sfIyK4LwgGg$eh4fG7MCsOPdGhAfB+r878peTifSN2DGZ4OkByirYpg
```

Then I use it in my `docker run` command like so

```
docker run -d --name codeserver --restart always -e HASHED_PASSWORD="$argon2i$v=19$m=4096,t=3,p=1$pYZYdaYLyi6sfIyK4LwgGg$eh4fG7MCsOPdGhAfB+r878peTifSN2DGZ4OkByirYpg" --network host linuxserver/code-server
```

**Comment by jsjoeio** [comment]
Hmm...I tested locally and also got an error. But if you set `hashed-password: "argon2..." it works.

Can you try that? I'll open a bug report though.

**Reply by jsjoeio** [reply]
reported here: https://github.com/coder/code-server/issues/4691

**Reply by simonjcarr** [reply]
Thank you and much appreciated, at least I know it's not be going mad.

**Reply by jsjoeio** [reply]
Hahaha yes, sorry we didn't get back sooner. Happy it's a bug and not us going crazy 😂
